staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -106,10 +107,23 @@ func PatchResource(r rest.Patcher, scope RequestScope, admit admission.Interface
 		)
 
 		userInfo, _ := request.UserFrom(ctx)
-		staticAdmissionAttributes := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo)
 		admissionCheck := func(updatedObject runtime.Object, currentObject runtime.Object) error {
-			if mutatingAdmission, ok := admit.(admission.MutationInterface); ok && admit.Handles(admission.Update) {
-				return mutatingAdmission.Admit(admission.NewAttributesRecord(updatedObject, currentObject, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
+			if currentObject == nil {
+				return fmt.Errorf("unexpected nil object in admission check of PATCH request")
+			}
+			if mutatingAdmission, ok := admit.(admission.MutationInterface); ok {
+				if mutatingAdmission.Handles(admission.Create) {
+					accessor, err := meta.Accessor(currentObject)
```

Comment:
I had assume that the currentObject is nil on create-on-update. Isn't that true? Then the accessor will fail.

Reply:
Or is it just a new+fresh object?

Reply:
[https://github.com/kubernetes/kubernetes/blob/5424061885901e9c5d538f879e930765ba4ab014/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L682](https://github.com/kubernetes/kubernetes/blob/5424061885901e9c5d538f879e930765ba4ab014/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L682)

@sttts It depends on whether we set `ignoreNotFound`.😅It could either be nil value or empty object.

Reply:
but in general we will see the fresh object on the handler+registry level, right? I.e. no nil.

Reply:
@sttts Generally, yes. Except for creation for CRD. It protects server's storage from abusing CRD CREATE requests which is very dangerous.

[https://github.com/kubernetes/kubernetes/blob/769b7dadcada0cf8a18739027b85f0676c773870/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/etcd.go#L105](https://github.com/kubernetes/kubernetes/blob/769b7dadcada0cf8a18739027b85f0676c773870/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/etcd.go#L105)

Reply:
> Except for creation for CRD

No sure I can follow. The linked code is about the CRD storage delete implementation (adding the finalizer).

Reply:
> Not sure if CRD creation will reach this call site. 

@nikhita Appreciated if you could shed some light on this 😃

Reply:
> @nikhita Appreciated if you could shed some light about this 

I think this change has been reverted?

> Not sure if CRD creation will reach this call site.

Can you elaborate on what the "call site" is here? :)

Reply:
@nikhita Thanks a ton for reviewing!

>> Not sure if CRD creation will reach this call site.
> Can you elaborate on what the "call site" is here? :)

Literally, I mean the line of code where `GuaranteedUpdate` is in the linked above.😃

> I might be totally wrong here :) but if the object is being "created" here, won't oldObj be always nil?

Addressing your comment below here, `oldObj` might either be nil or a zero-valued struct and it depends on whether set `ignoreNotFound` when calling `GuaranteedUpdate`. Currently searching through the project, the only place where `GuaranteedUpdate` is invoked with a false `ignoreNotFound` is the linked code above in which we append finalizer on CRD deletion. Do you think it's reasonable to check whether `oldObj` is nil here?